add 3d input and output shapes to maxpool documentation

migeed-z opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61790
* #61776
* #61747
* **#61310**

Differential Revision: [D29737516](https://our.internmc.facebook.com/intern/diff/D29737516)

migeed-z committed:
add 3d input and output shapes to maxpool documentation

[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

jbschlosser commented on torch/nn/modules/pooling.py:
```diff
@@ -95,7 +95,8 @@ class MaxPool2d(_MaxPoolNd):
 
     In the simplest case, the output value of the layer with input size :math:`(N, C, H, W)`,
-    output :math:`(N, C, H_{out}, W_{out})` and :attr:`kernel_size` :math:`(kH, kW)`
+    output :math:`(N, C, H_{out}, W_{out})` or with input size :math:`(C, H, W)`,
```
Note that the update here doesn't fit with the math below, as the math references `N_i`. I'd suggest leaving this part as-is.

jbschlosser commented on torch/nn/modules/pooling.py:
```diff
@@ -140,6 +144,7 @@ class MaxPool2d(_MaxPoolNd):
               W_{out} = \left\lfloor\frac{W_{in} + 2 * \text{padding[1]} - \text{dilation[1]}
                     \times (\text{kernel\_size[1]} - 1) - 1}{\text{stride[1]}} + 1\right\rfloor
 
+
```
nit: any need for the extra newline?

jbschlosser commented on torch/nn/modules/pooling.py:
```diff
@@ -130,7 +131,10 @@ class MaxPool2d(_MaxPoolNd):
     Shape:
         - Input: :math:`(N, C, H_{in}, W_{in})`
-        - Output: :math:`(N, C, H_{out}, W_{out})`, where
+        - Output: :math:`(N, C, H_{out}, W_{out})` or
```
For consistency with other modules, both possible shapes should be on a single Input and Output line (check out [nn.Upsample](https://pytorch.org/docs/stable/generated/torch.nn.Upsample.html) as an example).

jbschlosser reviewed (commented):
Thanks for the update! I added a couple nitpicky comments below, but we definitely want the 3D case documented :)

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"




[ghstack-poisoned]

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"


Differential Revision: [D29737516](https://our.internmc.facebook.com/intern/diff/D29737516)

[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"


Differential Revision: [D29737516](https://our.internmc.facebook.com/intern/diff/D29737516)

[ghstack-poisoned]

jbschlosser commented on torch/nn/modules/pooling.py:
```diff
@@ -129,8 +129,8 @@ class MaxPool2d(_MaxPoolNd):
-        - Input: :math:`(N, C, H_{in}, W_{in})`
-        - Output: :math:`(N, C, H_{out}, W_{out})`, where
+        - Input: :math:`(N, C, H_{in}, W_{in})` or :math:`(C, H_{in}, W_{in})`
+        - Output: :math:`(N, C, H_{out}, W_{out})` or :math:`(C, H_{out}, W_{out})` where,
```
nit: comma should still be before the where

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"


Differential Revision: [D29737516](https://our.internmc.facebook.com/intern/diff/D29737516)

[ghstack-poisoned]

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"


Differential Revision: [D29737516](https://our.internmc.facebook.com/intern/diff/D29737516)

[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"


Differential Revision: [D29737516](https://our.internmc.facebook.com/intern/diff/D29737516)

[ghstack-poisoned]

jbschlosser reviewed (approved):
LGTM!

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"


Differential Revision: [D29737516](https://our.internmc.facebook.com/intern/diff/D29737516)

[ghstack-poisoned]

migeed-z committed:
Update on "add 3d input and output shapes to maxpool documentation"


Differential Revision: [D29737516](https://our.internmc.facebook.com/intern/diff/D29737516)

[ghstack-poisoned]

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29737516).
